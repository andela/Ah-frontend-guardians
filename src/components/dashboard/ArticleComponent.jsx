import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { NavLink } from 'react-router-dom';
import default_image from '../../images/default_image.jpeg';

export const Article = (props) => {
  const getPlainText = data => data.replace(/<(?:.|\n)*?>/gm, '');
  const slug = props.article && props.article.slug;
  const url = 'article/' + slug;

  return (
    <NavLink to={url} id="Nav">
      <div className="article-box welocme_msg">
        <div>
          {' '}
          <img id="articleImage" src={default_image} width="300" height="150" />
        </div>
        <div id="cont">
          <h3>{props.article && props.article.title}</h3>
          <p className="text-dark">
            {props.article && props.article.description.substring(0, 50) + '...'}
          </p>
          <div>
            by <font color="red">{props.article && props.article.author.username}</font>
            &nbsp;| &nbsp;
            <font style={{ color: `${'#25ABD1'}` }}>
              {props.article && props.article.read_time + ' min read'}
            </font>
          </div>
        </div>
      </div>
    </NavLink>
  );
};
Article.propTypes = {
  article: PropTypes.shape({
    author: PropTypes.object,
    body: PropTypes.string,
    title: PropTypes.string,
    created_at: PropTypes.string
  })
};
export default Article;
