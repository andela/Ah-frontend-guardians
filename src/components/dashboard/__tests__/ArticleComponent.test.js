import React from 'react';
import {shallow} from 'enzyme';
import {Article} from '../ArticleComponent';

const props = {
    article:{title: "",
            author: "",
            description: "",
            created_at: "",
            image_url: "",
            slug: "",
        }
    }
describe('Test Article Component', ()=>{
    it('should render Article component correctly', ()=>{
        let wrapper = shallow(<Article {...props}/>);
        expect(wrapper).toMatchSnapshot();

    });
    it('should render article component correctly', ()=>{
        // let wrapper = shallow(<Article {...props}/>);
        // console.log(wrapper)
        // const findNavLink = wrapper.dive().find('NavLink');
        // expect(findNavLink.length).toEqual(1);
    });


})