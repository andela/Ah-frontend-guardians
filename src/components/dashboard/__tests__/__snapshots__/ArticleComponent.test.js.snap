// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Article Component should render Article component correctly 1`] = `ShallowWrapper {}`;
